### PR TITLE
MQTT Client TLS Implementation

### PR DESCRIPTION
**Pull request by eye0fra**
Extends the MqttConnector to allow TLS configuration on the MQTT Client to Address issue https://github.com/smallrye/smallrye-reactive-messaging/issues/693.

```
@ConnectorAttribute(name = "ssl.keystore.type", type = "string", direction = INCOMING_AND_OUTGOING, description = "Set the keystore type", defaultValue = "pkcs12")
@ConnectorAttribute(name = "ssl.keystore.location", type = "string", direction = INCOMING_AND_OUTGOING, description = "Set the keystore location")
@ConnectorAttribute(name = "ssl.keystore.password", type = "string", direction = INCOMING_AND_OUTGOING, description = "Set the keystore password")
@ConnectorAttribute(name = "ssl.truststore.type", type = "string", direction = INCOMING_AND_OUTGOING, description = "Set the truststore type", defaultValue = "pkcs12")
@ConnectorAttribute(name = "ssl.truststore.location", type = "string", direction = INCOMING_AND_OUTGOING, description = "Set the truststore location")
@ConnectorAttribute(name = "ssl.truststore.password", type = "string", direction = INCOMING_AND_OUTGOING, description = "Set the truststore password")
```

To implement:

- [x] Code Best Practices of Smallrye
- [x] PEM Implementation
- [x] One Way TLS Test Case
- [x] Mutual TLS Test Case
- [x] Documentation

### CODE REVIEW

File: smallrye-reactive-messaging-mqtt/src/test/resources/mosquitto-tls/client/client.crt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+-----BEGIN CERTIFICATE-----
```
Comment (cescoffier) [inline smallrye-reactive-messaging-mqtt/src/test/resources/mosquitto-tls/client/client.crt:1]: Please add a README with the generation instructions. I've been too many times bitten by expiration.

File: smallrye-reactive-messaging-mqtt/src/main/java/io/smallrye/reactive/messaging/mqtt/MqttHelpers.java [hunk 2]
```diff
@@ -24,11 +28,71 @@ static MqttClientOptions createMqttClientOptions(MqttConnectorCommonConfiguratio
...
+     * Description: Set whether keystore type, location and password
+     * Default Value: PfxOptions
+     * 
+     * @return the KeyCertOptions
+     */
+    private static KeyCertOptions getKeyCertOptions(MqttConnectorCommonConfiguration config) {
+
+        if (config.getSsl() && config.getSslKeystoreLocation().isPresent()) {
+            String keyStoreLocation = config.getSslKeystoreLocation().get();
+            String keyStorePassword = config.getSslKeystorePassword().get();
```
Comment (cescoffier) [inline smallrye-reactive-messaging-mqtt/src/main/java/io/smallrye/reactive/messaging/mqtt/MqttHelpers.java:54]: You probably want to catch when the key store is set and not the password.

File: smallrye-reactive-messaging-mqtt/src/main/java/io/smallrye/reactive/messaging/mqtt/MqttHelpers.java [hunk 3]
```diff
@@ -24,11 +28,71 @@ static MqttClientOptions createMqttClientOptions(MqttConnectorCommonConfiguratio
...
+            // enum? No enum available in Vertx MQTT client
```
Comment (cescoffier) [inline smallrye-reactive-messaging-mqtt/src/main/java/io/smallrye/reactive/messaging/mqtt/MqttHelpers.java:55]: yeah.. no enum :-(

File: smallrye-reactive-messaging-mqtt/src/main/java/io/smallrye/reactive/messaging/mqtt/MqttHelpers.java [hunk 4]
```diff
@@ -24,11 +28,71 @@ static MqttClientOptions createMqttClientOptions(MqttConnectorCommonConfiguratio
...
+     * Default Value: PfxOptions
+     * 
+     * @return the TrustOptions
+     */
+
+    private static TrustOptions getTrustOptions(MqttConnectorCommonConfiguration config) {
+
+        if (config.getSsl() && config.getSslTruststoreLocation().isPresent()) {
+            String keyStoreLocation = config.getSslTruststoreLocation().get();
+            String keyStorePassword = config.getSslTruststorePassword().get();
```
Comment (cescoffier) [inline smallrye-reactive-messaging-mqtt/src/main/java/io/smallrye/reactive/messaging/mqtt/MqttHelpers.java:83]: Make sure you catch the case where the user didn't pass the password.

File: smallrye-reactive-messaging-mqtt/src/test/resources/mosquitto-tls/client/client.crt (hunk 1 above)
Comment (eye0fra) [reply smallrye-reactive-messaging-mqtt/src/test/resources/mosquitto-tls/client/client.crt:1]: Where should I put the README?

Comment (cescoffier) [reply smallrye-reactive-messaging-mqtt/src/test/resources/mosquitto-tls/client/client.crt:1]: In sec/test/resources

Comment (eye0fra) [reply smallrye-reactive-messaging-mqtt/src/test/resources/mosquitto-tls/client/client.crt:1]: Ok, I put it under the mosquitto-tls directory.

**Review by cescoffier — APPROVED** [review]
That's awesome!
Thank you! 

Really appreciate the effort on the tests!

**Comment by cescoffier** [comment]
Created Backports: 
- #1126 to [backport-#1120-to-3.x](https://github.com/smallrye/smallrye-reactive-messaging/tree/3.x)

**Pull request merged**